### PR TITLE
ui v2: restyle metadata table

### PR DESCRIPTION
**Pull request by dschaller**
### Description

Restyle metadata table.

Screen Shot 2020-12-01 at 8 49 59 PM

### Testing Performed

storybook

### CODE REVIEW

File: frontend/packages/core/src/Table/metadata-table.tsx [hunk 1]
```diff
@@ -145,22 +210,29 @@ const MetadataTable: React.FC<MetadataTableProps> = ({
...
+      <Table>
         <TableBody>
-          {rows.map((row: IdentifiableRowData) => {
+          {rows.map((row: IdentifiableRowData, idx: number) => {
+            let position = "middle" as RowPosition;
+            if (idx === 0) {
+              position = "first";
+            } else if (idx === rows.length - 1) {
+              position = "last";
+            }
```
Comment (danielhochman) [inline frontend/packages/core/src/Table/metadata-table.tsx:223]: can't we do this with css selectors `:first-child` and `:last-child`?

File: frontend/packages/core/src/Table/metadata-table.tsx [hunk 2]
```diff
@@ -103,18 +137,17 @@ const EditableRow: React.FC<EditableRowProps> = ({
   );
 };
 
-interface MetadataTableProps {
+export interface MetadataTableProps {
   data: RowData[];
   onUpdate?: (id: string, value: unknown) => void;
   variant?: Size;
```
Comment (danielhochman) [inline frontend/packages/core/src/Table/metadata-table.tsx:143]: is there a use case for this prop? i didn't see it as part of any of the existing designs

File: frontend/packages/core/src/Table/metadata-table.tsx [hunk 3]
```diff
@@ -26,33 +32,69 @@ interface IdentifiableRowData extends RowData {
...
+const KeyCellContainer = styled(TableCell)({
+  width: "45%",
+  background: "rgba(13, 16, 48, 0.03)",
+  fontWeight: 500,
+});
+
 interface KeyCellProps {
   data: IdentifiableRowData;
   size: Size;
+  isLast?: boolean;
```
Comment (danielhochman) [inline frontend/packages/core/src/Table/metadata-table.tsx:84]: i think this is obsolete now

```suggestion
```

File: frontend/packages/core/src/Table/metadata-table.tsx [hunk 4]
```diff
@@ -26,33 +32,69 @@ interface IdentifiableRowData extends RowData {
...
+    "*:first-child": {
+      borderBottomLeftRadius: "3px",
+    },
+    "*:last-child": {
+      borderBottomRightRadius: "3px",
+    },
+  },
+});
+
+const TableCell = styled(MuiTableCell)({
```
Comment (danielhochman) [inline frontend/packages/core/src/Table/metadata-table.tsx:69]: figma cells are ~5px less in height than these

File: frontend/packages/core/src/Table/stories/metadata-table.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        }}
+      >
+        {story()}
+      </WizardContext.Provider>
+    ),
+  ],
+  component: MetadataTable,
+} as Meta;
+
+const Template = (props: MetadataTableProps) => (
```
Comment (danielhochman) [inline frontend/packages/core/src/Table/stories/metadata-table.stories.tsx:30]: this is rendering in storybook code preview as `<[object Object] value={() => {}}>` for some reason

**Review by danielhochman — COMMENTED** [review]
nits

File: frontend/packages/core/src/Table/metadata-table.tsx (hunk 2 above)
Comment (dschaller) [reply frontend/packages/core/src/Table/metadata-table.tsx:143]: I wasn't sure if we should keep it or not. My main thought was if we wanted to embed this within a full page dashboard and make it smaller for informational purposes. I can remove for now though.

File: frontend/packages/core/src/Table/stories/metadata-table.stories.tsx (hunk 5 above)
Comment (danielhochman) [reply frontend/packages/core/src/Table/stories/metadata-table.stories.tsx:30]: oh something to do with the decorator?

Comment (dschaller) [reply frontend/packages/core/src/Table/stories/metadata-table.stories.tsx:30]: This is the application context wrapper. It looks like the context provider is rendering as the return value (e.g. an object in this case).

Comment (danielhochman) [reply frontend/packages/core/src/Table/stories/metadata-table.stories.tsx:30]: ok not a blocker

**Review by danielhochman — APPROVED** [review]

**Pull request merged**